app/controllers/general.php

```diff
@@ -155,6 +156,70 @@ function router(App $utopia, Database $dbForPlatform, callable $getProjectDB, Sw
             $path .= '?' . $query;
         }
 
+        $protocol = $request->getProtocol();
+
+        // Ensure preview authorization
+        $requirePreview = \is_null($apiKey) || !$apiKey->isPreviewAuthDisabled();
+        if ($isPreview && $requirePreview) {
+            $cookie = $request->getCookie(Auth::$cookieNamePreview, '');
+            $authorized = false;
+
+            // Security checks to mark authorized true
+            if (!empty($cookie)) {
+                $jwt = new JWT(System::getEnv('_APP_OPENSSL_KEY_V1'), 'HS256', 3600, 0);
```

Comment:
Since this is going to be part of the edge, with no access to the database, can we instead redirect this to another Appwrite endpoint that will check the DB, issue the token, and redirect back so we can set it as a local cookie?

Reply:
Agree, let's address later when edge is finished